File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        if head is None:
            return None
        value=[]
        node=[]
        ptr=head
        c=0
        while ptr !=None:
            c+=1
            if c>=left and c<=right:
                value.append(ptr.val)
                node.append(ptr)
            ptr=ptr.next
        for i in node:
            i.val=value.pop(-1)
        return head

File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseBetween(ListNode head, int left, int right) {
        if(head==null) {
            return null;
        }
        ListNode[] value=new ListNode[right-left+1];
        ListNode[] node=new ListNode[right-left+1];
        ListNode ptr=head;
        int c=0;
        while(ptr!=null) {
            c++;
            if(c>=left && c<=right){
                value[c-left]=ptr;
                node[c-left]=new ListNode(ptr.val);
            }
            ptr=ptr.next;
        }
        for(int i=0; i<node.length; i++){
            value[i].val=node[node.length-1-i].val;
        }
        return head;
    }
}